src/internal/SetBuildNumber/SetBuildNumber.proj

```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the Microsoft Reciprocal License. See LICENSE.TXT file in the project root for full license information. -->
+
+<Project Sdk="Microsoft.Build.NoTargets/3.0.4">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <!-- Disable central package version management since this project will write the central package version management file -->
+    <ManagePackageVersionsCentrally>false</ManagePackageVersionsCentrally>
+
+    <!-- Explicitly set the restore sources because this project may run before the build\artifacts folder has been created -->
+    <RestoreSources>https://api.nuget.org/v3/index.json</RestoreSources>
+    <RestoreIgnoreFailedSources>true</RestoreIgnoreFailedSources>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <SetBuildNumbersDependsOn>
+    __SetPropertiesFromGit;
+      SetAppVeyorBuildNumber;
+      SetGlobalJson;
+      SetDirectoryPackagesProps;
+    </SetBuildNumbersDependsOn>
+
+    <GlobalJsonPath>$([System.IO.Path]::GetFullPath($(MSBuildThisFileDirectory)..\..\global.json))</GlobalJsonPath>
+    <CentralPackageVersionsPath>$([System.IO.Path]::GetFullPath($(MSBuildThisFileDirectory)..\..\Directory.Packages.props))</CentralPackageVersionsPath>
+  </PropertyGroup>
+
+  <Target Name="SetAppVeyorBuildNumber"
+          Condition=" '$(APPVEYOR)'=='True' ">
+    <PropertyGroup>
+      <AppVeyorBuildNumber>$(InformationalVersion)</AppVeyorBuildNumber>
+      <AppVeyorBuildNumber Condition=" 'APPVEYOR_PULL_REQUEST_NUMBER'!='' ">PR#$(APPVEYOR_PULL_REQUEST_NUMBER)-$(AppVeyorBuildNumber)</AppVeyorBuildNumber>
```

Comment:
I meant to fix this in my PR and forgot. Needs to be `Condition=" '$(APPVEYOR_PULL_REQUEST_NUMBER)'!='' "`, it's missing `$()`.

Reply:
Yep. I have that fixed locally with another change. Coming soon.